packages/elements/src/interactive-chart/helpers/types.ts

```diff
@@ -1,40 +1,23 @@
 import type { HSLColor, RGBColor } from '@refinitiv-ui/utils/color.js';
 import type {
-  AreaSeriesPartialOptions,
-  AreaStyleOptions,
-  BarData,
-  BarSeriesPartialOptions,
-  BarStyleOptions,
-  CandlestickSeriesPartialOptions,
-  CandlestickStyleOptions,
   ChartOptions,
   DeepPartial,
-  HistogramData,
-  HistogramSeriesPartialOptions,
-  HistogramStyleOptions,
   ISeriesApi,
-  LineData,
-  LineSeriesPartialOptions,
-  LineStyleOptions,
-  SeriesPartialOptions,
+  SeriesDataItemTypeMap,
+  SeriesOptionsMap,
   SeriesType
 } from 'lightweight-charts';
 
-type SeriesOptions =
-  | AreaSeriesPartialOptions
-  | BarSeriesPartialOptions
-  | CandlestickSeriesPartialOptions
-  | HistogramSeriesPartialOptions
-  | LineSeriesPartialOptions;
-type SeriesStyleOptions = LineStyleOptions &
-  AreaStyleOptions &
-  BarStyleOptions &
-  CandlestickStyleOptions &
-  HistogramStyleOptions;
+// https://stackoverflow.com/a/50375286
+type UnionToIntersection<U> = (U extends unknown ? (k: U) => void : never) extends (k: infer I) => void
+  ? I
+  : never;
+
+type SeriesOptions = UnionToIntersection<SeriesOptionsMap[keyof SeriesOptionsMap]>;
 
-type SeriesData = LineData[] | BarData[] | HistogramData[];
 type SeriesList = ISeriesApi<SeriesType>;
-type SeriesDataItem = BarData | LineData;
+
```

Comment:
let's add a comment. This's not that straightforward.
```suggestion
// a type union of each prop in `SeriesDataItemTypeMap` interface i.e. `BarData |  CandlestickData | ...`
```